src/CoreFoundation/CFProxySupport.cs

```diff
@@ -580,16 +580,23 @@ public static CFProxySettings GetSystemProxySettings ()
 			return new CFProxySettings (dict);
 		}
 		
-#if notyet
-		// FIXME: These require bindings for CFRunLoopSource and CFStreamClientContext
-		
 		public delegate void CFProxyAutoConfigurationResultCallback (IntPtr client, NSArray proxyList, NSError error);
```

Comment:
@mandel-macaque and @rolfbjarne - is it possible for this callback to return a C# array of CFProxy rather than an NSArray? It would be quite a bit more convenient from the client side if we didn't have to parse the NSArray into the CFProxy array. Also, I've been testing this API with @mandel-macaque and am unsure how to parse the NSArray into the C# array of CFProxy - is this possible? Can one of you provide some guidance on this front?

Reply:
We should be able to make it a managed array of something, but that something would probably be typed `NSObject`, since Apple doesn't explicitly state the type of the elements in the array. Then at runtime you can do a standard C# type check to check the actual type of the elements you get.